pallets/asset-manager/src/lib.rs

```diff
@@ -214,6 +216,75 @@ pub mod pallet {
 	#[pallet::storage]
 	pub type UnitsPerSecond<T: Config> = StorageMap<_, Blake2_128Concat, AssetId, u128>;
 
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_runtime_upgrade() -> Weight {
+			if have_storage_value(Self::name().as_bytes(), b"NextAssetId", &[]) {
+				log::warn!(" !!! Aborting, NextAssetId was already set.");
+				return T::DbWeight::get().reads_writes(1, 0);
+			}
+
+			let asset_id = <T::AssetConfig as AssetConfig<T>>::NativeAssetId::get();
+			let metadata = <T::AssetConfig as AssetConfig<T>>::NativeAssetMetadata::get();
+			let location = <T::AssetConfig as AssetConfig<T>>::NativeAssetLocation::get();
+			if have_storage_value(
+				Self::name().as_bytes(),
+				b"AssetIdLocation",
+				&Blake2_128Concat::hash(&asset_id.encode()),
+			) {
+				log::warn!(" !!! Aborting, AssetIdLocation was already set.");
+				return T::DbWeight::get().reads_writes(4, 0);
```

Comment:
you should be at 5 reads here ( and +1 at each one below )

Reply:
Check https://github.com/paritytech/cumulus/blob/master/pallets/parachain-system/src/migration.rs#L37
for how parity avoid messing up the counting

Reply:
Should be resolved now.